modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]DefineAPISchemasWithinAnAPIGroup/CreateAPIApplicationSchema.testcase

```diff
@@ -0,0 +1,207 @@
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+		task ("Given New API Application created") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				status = "unpublished",
+				title = "My-app");
+
+			ApplicationAPI.setUpGlobalAPIApplicationId();
+		}
+
+		task ("And Given I go to the Schemas tab of API Application edition") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Schemas",
+				title = "My-app");
+		}
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		JSONObject.deleteAllCustomObjects(pageSize = 22);
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 5
+	test AllObjectDefinitionsAreAvailableInTheObjectMenu {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given multiple custom object definitions created without publishing") {
+			for (var index : list "1,2,3,4,5,6,7,8") {
+				ObjectDefinitionAPI.createObjectDefinition(
+					en_US_label = "Student${index}",
+					en_US_plural_label = "student${index}s",
+					name = "Student${index}",
+					requiredStringFieldName = "name");
+			}
+		}
+
+		task ("And Given all 20+ object deinifition entities available in Objects") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "Objects");
+
+			AssertTextEquals(
+				locator1 = "Pagination#RESULTS",
+				value1 = "Showing 1 to 20 of 22 entries.");
+		}
+
+		task ("And Given one of object definitions on 2nd page") {
+			Click(locator1 = "DigitalSignatureListView#DIGITAL_SIGNATURE_NAVIGATE_NEXT");
+
+			AssertElementPresent(
+				key_name = "Student8",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+		}
+
+		task ("And Given I go to the Schemas tab of API Application edition") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Schemas",
+				title = "My-app");
+		}
+
+		task ("And Given I click Add New Schema") {
+			Click(locator1 = "APIBuilder#ADD_NEW_SCHEMA");
+		}
+
+		task ("When I click Object field") {
+			Click(locator1 = "APIBuilder#SELECT_AN_OBJECT_DEFINITION");
+		}
+
+		task ("Then all object deinifition entities from Objects are listed inluding the one placed on 2nd page of Objects") {
+			for (var objectDefinitionName : list "Address,AccountEntry,Organization,User,CommercePricingClass,CPDefinition,CommerceOrder,APISort,Bookmark,APIFilter,APIEndpoint,APIProperty,APISchema,APIApplication,Student1,Student2,Student3,Student4,Student5,Student6,Student7,Student8") {
```

Comment:
Are we able to split this line into a few shorter ones? 🤔  Not sure if it's possible here, but it looks kinda ugly all of it together :D Like a bit unformatted :D